### PR TITLE
Use sysfs to check interface state rather than shelling out.

### PR DESCRIPTION
**Pull request by fasaxc**
This is much faster.

Also, check the operstate, which matches what the watch loop in devices.py checks.

### CODE REVIEW

**Comment by fasaxc** [comment]
retest this please

**Comment by fasaxc** [comment]
@plwhite OK, now it's assigned to you!

**Comment by fasaxc** [comment]
retest this please

File: calico/felix/devices.py [hunk 1]
```diff
@@ -264,27 +254,25 @@ def interface_up(if_name):
...
-        _log.warning("Cannot check flags for interface %s (%s) - assume "
-                     "down: %r.", if_name, flags_file, e)
+        _log.warning("Failed to read state of interface %s (%s) - assume "
+                     "down/absent: %r.", if_name, operstate_filename, e)
         return False
-
-    return bool(int(flags, 16) & 1)
+    else:
+        _log.debug("Interface %s has state %s", if_name, oper_state)
+    return oper_state == "up"
```
Comment (plwhite) [inline calico/felix/devices.py]: Can we have a (probably debug level) log saying what we read please.

**Comment by plwhite** [comment]
LGTM. One trivial markup, which is as always complaining about logging.

File: calico/felix/devices.py (hunk 1 above)
Comment (alexwlchan) [inline calico/felix/devices.py]: I'm confused. Isn't that already logged on line 274?

Comment (plwhite) [inline calico/felix/devices.py]: Doh. Yes, you are indeed correct. I have no idea how I managed to miss that, given that it's on the next line to the one I was looking at. I withdraw my objection and shall return to my slumber.

**Pull request merged**